ts/Stock/Navigator/StandaloneNavigator.ts

```diff
@@ -0,0 +1,330 @@
+    }
+}
+
+type StandaloneNavigatorOptions = {
+    navigator: NavigatorOptions;
+    width: number;
+    height: number;
+    min: number;
+    max: number;
+    colors: [];
+};
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The StandaloneNavigator class.
+ *
+ * @class
+ * @name Highcharts.StandaloneNavigator
+ *
+ * @param {string|Highcharts.HTMLDOMElement} [renderTo]
+ * The DOM element to render to, or its id.
+ *
+ * @param {DeepPartial<StandaloneNavigatorOptions>} userOptions
+ * The standalone navigator options with chart-like structure.
+ */
+class StandaloneNavigator {
+
+    public eventsToUnbind: Array<Function> = [];
+    public navigator: Navigator;
+    public boundAxes: Array<Axis> = [];
+    public options: StandaloneNavigatorOptions;
+    public userOptions: DeepPartial<StandaloneNavigatorOptions>;
+
+    /* *
+     *
+     *  Static Functions
+     *
+     * */
+
+    /**
+     * Factory function for standalone navigator.
+     *
+     * @function Highcharts.navigator
+     *
+     * @param {string|Highcharts.HTMLDOMElement} [renderTo]
+     * The DOM element to render to, or its id.
+     *
+     * @param {StandaloneNavigatorOptions} options
+     * The standalone navigator options with chart-like structure.
+     *
+     * Returns the navigator object.
+     */
+    public static navigator(
+        renderTo: (string|globalThis.HTMLElement),
+        options: DeepPartial<StandaloneNavigatorOptions>
+    ): StandaloneNavigator {
+        const nav = new StandaloneNavigator(renderTo, options);
+
+        if (!G.navigators) {
+            G.navigators = [nav];
+        } else {
+            G.navigators.push(nav);
+        }
+
+        return nav;
+    }
+
+
+    /* *
+     *
+     *  Constructor
+     *
+     * */
+
+    constructor(
+        element: (string | globalThis.HTMLElement),
+        userOptions: DeepPartial<StandaloneNavigatorOptions>
+    ) {
+        this.userOptions = userOptions;
+        this.options = merge(
+            (G as any).getOptions(),
+            StandaloneNavigatorDefaults,
+            userOptions
+        ) as StandaloneNavigatorOptions;
+
+        const chart = new Chart(element, this.options);
+
+        this.navigator = new Navigator(chart);
+        chart.navigator = this.navigator;
+        this.initNavigator();
+    }
+
+    /**
+     * Binds an axis to the standalone navigator,
+     * allowing the navigator to control the range.
+     *
+     * @param {Axis | Chart} axisOrChart
+     *        The Axis or Chart to bind to the navigator.
+     */
+    public bind(axisOrChart: Axis | Chart): void {
+        const nav = this;
+        // If the chart is passed, bind the first xAxis
+        const axis = (axisOrChart instanceof Chart) ?
+            axisOrChart.xAxis[0] :
+            axisOrChart;
+
+        if (!(axis instanceof Axis)) {
+            return;
+        }
+
+        const { min, max } = this.navigator.xAxis;
+
+        this.boundAxes.push(axis);
```

Comment:
We should bind to `setRange` event here. Instead of creating and maintaining `boundAxes` array. The same in `unbind` - just remove `setRange` event added here.

Then in `initNavigator()` you can remove the artificial event added to the chart's empty xAxis `setExtremes`

Reply:
Imo this solution is easier, because when you are keeping the reference to the correct object. The unbind is easier, because you are just deleting the correct object from the array. When unbinding the chart/axis, and passing it as an argument, how to find the correct event to remove, without mutating the bound axis?

Reply:
With `addEvent` and `removeEvent` you don't need a bound axis at all. I was thinking about something like this:

```js
bind(axis) {
  addEvent(this, 'setRange', axis.setExtremes);
}

unbind(axis) {
  removeEvent(this, 'setRange', axis.setExtremes);
}
```

But it seems I made a mistake in the draft, where I assumed compatible interfaces, but the Event interface is an object, not an Array. We could extend Axis prototype with an adapter:
```js
Axis.proto.setRange = function () {
  return this.setExtremes(...args);
}
```

And use it like I said above. Then we can get rid of this part of code: https://github.com/highcharts/highcharts/pull/20159/files#diff-900ccb90f094a9e5757b539c60572a4d491791cadcd1c824d1a56f4ac45a38b9R279-R291

What do you think?

Reply:
Hm, I didn't think of implementing it this way. Indeed it seems much simpler. Let's go for that